client/client.go

```diff
@@ -955,6 +896,12 @@ func (c *Client) updateNodeStatus() error {
 	defer c.heartbeatLock.Unlock()
 	c.lastHeartbeat = time.Now()
 	c.heartbeatTTL = resp.HeartbeatTTL
+
+	if err := c.rpcProxy.UpdateFromNodeUpdateResponse(&resp); err != nil {
```

Comment:
Done with one caveat: the argument for the `RefreshServerList()` is still an array of `struct.NodeServerInfo` pointers.  The goal of addressing the layering violation has been mostly achieved (or at least pushed down), but this still isn't "ideal."  Repackaging `structs.NodeServerInfo` for the sake of not reusing a struct intended for the API layer is puritanically correct, but not pragmatic atm.  LMK if you want me to revisit, but I think this is okay for now until the needs of the structure diverge.  For reference, here's what's in `NodeServerInfo`.

https://github.com/hashicorp/nomad/blob/bae7c72973c3bde15db87a8e469a2a083a935b56/nomad/structs/structs.go#L169